### PR TITLE
Move HTTP and Context into routeHandlers.

### PR DESCRIPTION
**Pull request by hhenrichsen**
This is still a work in progress, but I'm creating this so that it can be tracked more easily.

To Do:
- [x] Move `Response` into `routeHandlers`
- [x] Refactor `login`
- [x] Refactor `newGame`
- [x] Refactor `startGame`
- [x] Refactor `update`
- [x] Refactor `play`
- [x] Refactor `draw`

Potentially (pending feedback):
- Convert from POST with URL parameters on Play to POST with a body.

### CODE REVIEW

File: server/uno.go [hunk 1]
```diff
@@ -103,89 +95,60 @@ func contains(arr []string, val string) (int, bool) {
...
-	client := createClient(ctx)
-
-	_, _, err := client.Collection("users").Add(ctx, map[string]interface{}{
-		"first": "Ada",
-		"last":  "Lovelace",
-		"born":  1815,
-	})
-	if err != nil {
-		log.Fatalf("Failed adding alovelace: %v", err)
-	}
```
Comment (Aaron-Fine) [inline server/uno.go:126]: Is this to do with the database backend and that's why it's being removed?
If not, this should probably be part of the `routeHandlers.go` file.

File: server/uno.go [hunk 2]
```diff
@@ -103,89 +95,60 @@ func contains(arr []string, val string) (int, bool) {
...
+func playCard(game string, username string, card Card) bool {
+	if checkID(game) && currPlayer == username {
 		if card.Color == currCard[0].Color || card.Number == currCard[0].Number {
 			// Valid card can be played
 			playerIndex = (playerIndex + 1) % len(players)
 			currPlayer = players[playerIndex]
 			currCard[0] = card
 
-			for index, item := range allCards[c.Param("username")] {
+			for index, item := range allCards[username] {
```
Comment (Aaron-Fine) [inline server/uno.go:131]: Consider making `cards: := allCards[username]` as `allCards[username]` gets used  a lot here.

File: server/uno.go (hunk 1 above)
Comment (hhenrichsen) [reply server/uno.go]: Yeah I watched him remove this in class -- this was an experiment with the Firebase emulator I think.

File: server/uno.go (hunk 2 above)
Comment (hhenrichsen) [reply server/uno.go]: I've gone ahead and made that change as well.

**Comment by hhenrichsen** [comment]
One other thing we might consider is migrating the `newPayload` function out of `uno.go` into the route handlers. It's pretty closely tied to the Response type, which was also moved to the route handlers, but then the issue becomes getting access to the 'known variables' as they're commented in the code. Any thoughts on this?

**Comment by Aaron-Fine** [comment]
> One other thing we might consider is migrating the `newPayload` function out of `uno.go` into the route handlers. It's pretty closely tied to the Response type, which was also moved to the route handlers, but then the issue becomes getting access to the 'known variables' as they're commented in the code. Any thoughts on this?

I could see renaming the function to `getGameState` or something, but I think it will be very hard to extract this method. I also don't see much of a problem with it. Basically, it just returns an array containing the state of the current internal variables for the uno class. It seems self contained and easily testable...

If this data is mostly consumed by the front end, then perhaps the format of the response should be changed to JSON?

**Comment by hhenrichsen** [comment]
Yeah, I think we can just leave that then. Can I get a second review before we merge this?

**Comment by Aaron-Fine** [comment]
> Yeah, I think we can just leave that then. Can I get a second review before we merge this?

Everything looks good to me, but I thought the professor wanted to be the only one who merged things?

**Pull request merged**